quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/integration/web/QuarkusRequestFilter.java

```diff
@@ -87,20 +89,26 @@ public void handle(RoutingContext context) {
      * Creates a handler to close the {@link KeycloakSession} before the response is written to response but after Resteasy
      * is done with processing its output.
      */
-    private void configureEndHandler(RoutingContext context, Promise<Object> promise, KeycloakSession session) {
+    private void configureEndHandler(RoutingContext context, KeycloakSession session) {
         context.addHeadersEndHandler(event -> {
             try {
                 close(session);
             } catch (Throwable cause) {
-                promise.fail(cause);
-                context.response().headers().clear();
-                context.response().putHeader(HttpHeaderNames.CONTENT_TYPE, HttpHeaderValues.TEXT_PLAIN);
-                context.response().putHeader(HttpHeaderNames.CONTENT_LENGTH, "0");
-                context.response().setStatusCode(HttpResponseStatus.INTERNAL_SERVER_ERROR.code());
+                unexpectedErrorResponse(context.response());
             }
         });
     }
 
+    private void unexpectedErrorResponse(HttpServerResponse response) {
+        response.headers().clear();
+        response.putHeader(HttpHeaderNames.CONTENT_TYPE, HttpHeaderValues.TEXT_PLAIN);
+        response.putHeader(HttpHeaderNames.CONTENT_LENGTH, "0");
+        response.setStatusCode(HttpResponseStatus.INTERNAL_SERVER_ERROR.code());
+        response.putHeader(HttpHeaderNames.CONNECTION, HttpHeaderValues.CLOSE);
+        // writes an empty buffer to replace any data previously written
+        response.write(Buffer.buffer(""));
```

Comment:
```suggestion
        response.write(Buffer.buffer(""));
        response.end();
```

Make it sense to end the requests here? I guess that should prevent any additional writes on the request.

Reply:
I tried that and it seems unnecessary because that is done later by vert.x itself.